synapse/config/server.py

```diff
@@ -228,6 +228,90 @@ def read_config(self, config):
             "show_users_in_user_directory", True,
         )
 
+        retention_config = config.get("retention", {})
+
+        self.retention_enabled = retention_config.get("enabled", False)
+
+        self.retention_min_lifetime = self.parse_duration(
+            retention_config.get("min_lifetime", 0)
+        )
+
+        self.retention_max_lifetime = self.parse_duration(
+            retention_config.get("max_lifetime", 0)
+        )
+
+        if self.retention_enabled and not (
+            self.retention_min_lifetime and self.retention_max_lifetime
```

Comment:
is it deliberate to disallow setting these to zero?

Reply:
No